WordPress/Classes/ViewRelated/Reader/ReaderInterestsCoordinator.swift

```diff
@@ -0,0 +1,87 @@
+import Foundation
+
+class ReaderSelectInterestsCoordinator {
+    private struct Constants {
+        static let userDefaultsKeyFormat = "Reader.SelectInterests.hasSeenBefore.%@"
+    }
+
+    private let interestsService: ReaderFollowedInterestsService
+    private let store: KeyValueDatabase
+    private let userId: NSNumber?
+
+    /// Creates a new instance of the coordinator
+    /// - Parameter service: An Optional `ReaderFollowedInterestsService` to use. If this is `nil` one will be created on the main context
+    ///   - store: An optional backing store to keep track of if the user has seen the select interests view or not
+    ///   - userId: The logged in user account, this makes sure the tracking is a per-user basis
+    init(service: ReaderFollowedInterestsService? = nil,
+         store: KeyValueDatabase = UserDefaults.standard,
+         userId: NSNumber? = nil,
+         context: NSManagedObjectContext = ContextManager.sharedInstance().mainContext) {
+
+        self.interestsService = service ?? ReaderTopicService(managedObjectContext: context)
+        self.store = store
+        self.userId = userId ?? {
+            let acctServ = AccountService(managedObjectContext: context)
+            let account = acctServ.defaultWordPressComAccount()
+
+            return account?.userID
+        }()
+    }
+
+    // MARK: - Display Logic
+
+    /// Determines whether or not the select interests view should be displayed
+    /// - Returns: true 
+    public func shouldDisplay(completion: @escaping (Bool) -> Void) {
+        interestsService.fetchFollowedInterestsLocally { [weak self] (followedInterests) in
+            let shouldDisplay: Bool = self?.shouldDisplaySelectInterests(with: followedInterests) ?? false
+            completion(shouldDisplay)
+        }
+    }
+
+    private func shouldDisplaySelectInterests(with interests: [ReaderTagTopic]?) -> Bool {
+        guard let interests = interests else {
+            return false
+        }
+
+        return !hasSeenBefore() && interests.count <= 0
+    }
+
+    // MARK: - View Tracking
+
+    /// Generates the user defaults key for the logged in user
+    /// Returns nil if we can not get the default WP.com account
+    private var userDefaultsKey: String? {
+        get {
+            return String(format: Constants.userDefaultsKeyFormat, userId ?? 0)
```

Comment:
I'd rather guard for the userId and return nil in the else instead of `userID ?? 0` that way the code is self explanatory and I d on't have to wonder what will this return if userId is nil and we use 0

Reply:
If we return nil here the logged out user key will be nil which means the view tracking wouldn't be registered. 

I've updated how the default works by using `Constant.loggedOutUserKey` and removing the optional from the property: https://github.com/wordpress-mobile/WordPress-iOS/commit/2be6cdb2bdaac0a4d7efcb45a3aaa9a96c497d50